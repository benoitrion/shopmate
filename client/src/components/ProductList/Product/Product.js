import React from 'react';

import Card, {
  CardPrimaryContent,
  CardMedia,
  CardActions,
  CardActionButtons
} from '@material/react-card/'

import { Button } from '@material/react-button';
import MaterialIcon from '@material/react-material-icon';

import {
  Headline6,
  Body2
} from '@material/react-typography/'

import '@material/react-card/index.scss';
import '@material/react-typography/index.scss';
import '@material/react-button/index.scss';
import '@material/react-material-icon/index.scss';

import './Product.scss';

function Product({ product }) {
  const image = require(`../../../static/product_images/${product.image}`);
  return (
    <React.Fragment>
      <Card className='product-card'>
        <CardPrimaryContent>
          <CardMedia square imageUrl={image} alt={product.name} className='product-card__media' />
        </CardPrimaryContent>
        <div className='product-card__primary'>
          <Headline6 className='product-card__title'>
            {product.name}
          </Headline6>
        </div>
        <Body2 className='product-card__body'>
          {product.description}
        </Body2>
        <CardActions>
          <CardActionButtons>
            <Button className='product-card__button'>
              <MaterialIcon icon='add_shopping_cart' className='product-card__icon' />
              <span className="mdc-button__label">Add to cart</span>
            </Button>
          </CardActionButtons>
        </CardActions>
      </Card>
    </React.Fragment>
  );
}

export default Product;
