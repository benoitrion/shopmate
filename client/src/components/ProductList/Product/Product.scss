.product-card {
}
.product-card__media {
}
.product-card__primary {
}
.product-card__title {
  margin: 10px;
}
.product-card__body {
  height: 100px;
  margin: 10px;
}
.product-card__icon {
  margin-right: 5px;
  display: inline-block;
  vertical-align:middle;
}