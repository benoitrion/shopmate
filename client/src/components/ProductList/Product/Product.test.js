import { shallow, mount } from 'enzyme';
import React from 'react';
import Product from './Product';

describe('Header test', () => {
  const mockProduct = {
    id:"1",
    name: 'Arc d\'Triomphe',
    description: 'This beautiful and iconic T-shirt will no doubt lead you to your own triumph.',
    price: 14.99,
    discounted_price: 0.00,
    image: 'arc-d-triomphe.gif',
    image_2: 'arc-d-triomphe-2.gif',
    thumbnail: 'arc-d-triomphe-thumbnail.gif',
    display: 0
  } 
  let wrapper;
  beforeEach(() => wrapper = mount(<Product product={mockProduct}/>));
  
  it('should render correctly', () => {
    wrapper = shallow(<Product product={mockProduct}/>)
    expect(wrapper).toMatchSnapshot()
  });
  
  it('renders name and description components', () => {
    expect(wrapper.find('h6').text()).toEqual(mockProduct.name);
    expect(wrapper.find('p').text()).toEqual(mockProduct.description);
  });

  it('renders product', () => {
    expect(wrapper.props().product).toEqual(mockProduct);
  });

});