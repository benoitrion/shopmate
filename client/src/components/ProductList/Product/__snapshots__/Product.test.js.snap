// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render correctly 1`] = `
<Fragment>
  <Card
    className="product-card"
  >
    <RippledComponent
      className=""
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      unbounded={false}
    >
      <Media
        alt="Arc d'Triomphe"
        className="product-card__media"
        imageUrl="arc-d-triomphe.gif"
        square={true}
      />
    </RippledComponent>
    <div
      className="product-card__primary"
    >
      <Typography
        className="product-card__title"
      >
        Arc d'Triomphe
      </Typography>
    </div>
    <Typography
      className="product-card__body"
    >
      This beautiful and iconic T-shirt will no doubt lead you to your own triumph.
    </Typography>
    <Actions>
      <ActionButtons>
        <Component
          className="product-card__button"
          initRipple={[Function]}
        >
          <MaterialIcon
            className="product-card__icon"
            hasRipple={false}
            icon="add_shopping_cart"
            unbounded={false}
          />
          <span
            className="mdc-button__label"
          >
            Add to cart
          </span>
        </Component>
      </ActionButtons>
    </Actions>
  </Card>
</Fragment>
`;
